src/GraphQL.Tests/Types/AllScalarGraphTypeTests.cs

```diff
@@ -0,0 +1,807 @@
+        [InlineData(typeof(LongGraphType), false)]
+        [InlineData(typeof(ULongGraphType), false)]
+        [InlineData(typeof(BigIntGraphType), false)]
+        [InlineData(typeof(StringGraphType), 1.5)]
+        [InlineData(typeof(ByteGraphType), 1.5)]
+        [InlineData(typeof(SByteGraphType), 1.5)]
+        [InlineData(typeof(ShortGraphType), 1.5)]
+        [InlineData(typeof(UShortGraphType), 1.5)]
+        [InlineData(typeof(IntGraphType), 1.5)]
+        [InlineData(typeof(UIntGraphType), 1.5)]
+        [InlineData(typeof(LongGraphType), 1.5)]
+        [InlineData(typeof(ULongGraphType), 1.5)]
+        [InlineData(typeof(BigIntGraphType), 1.5)]
+        public void parseLiteral_other_fail(Type graphType, object value)
+        {
+            IValue astValue = value switch
+            {
+                int i => new IntValue(i),
+                long l => new LongValue(l),
+                bool b => new BooleanValue(b),
+                double d => new FloatValue(d),
+                string s => new StringValue(s),
+                _ => null
+            };
+
+            var g = (ScalarGraphType)graphType.GetConstructor(Type.EmptyTypes).Invoke(null);
+            Should.Throw<Exception>(() => g.ParseLiteral(astValue));
+            g.CanParseLiteral(astValue).ShouldBeFalse();
+        }
+
+        [Theory]
+        [InlineData(typeof(BooleanGraphType), 0)]
+        [InlineData(typeof(BooleanGraphType), 1)]
+        [InlineData(typeof(BooleanGraphType), "false")]
+        [InlineData(typeof(BooleanGraphType), "False")]
+        [InlineData(typeof(BooleanGraphType), "true")]
+        [InlineData(typeof(BooleanGraphType), "True")]
+        [InlineData(typeof(StringGraphType), 0)]
+        [InlineData(typeof(StringGraphType), false)]
+        [InlineData(typeof(ByteGraphType), false)]
+        [InlineData(typeof(SByteGraphType), false)]
+        [InlineData(typeof(ShortGraphType), false)]
+        [InlineData(typeof(UShortGraphType), false)]
+        [InlineData(typeof(IntGraphType), false)]
+        [InlineData(typeof(UIntGraphType), false)]
+        [InlineData(typeof(LongGraphType), false)]
+        [InlineData(typeof(ULongGraphType), false)]
+        [InlineData(typeof(BigIntGraphType), false)]
+        [InlineData(typeof(StringGraphType), 1.5)]
+        [InlineData(typeof(ByteGraphType), 1.5)]
+        [InlineData(typeof(SByteGraphType), 1.5)]
+        [InlineData(typeof(ShortGraphType), 1.5)]
+        [InlineData(typeof(UShortGraphType), 1.5)]
+        [InlineData(typeof(IntGraphType), 1.5)]
+        [InlineData(typeof(UIntGraphType), 1.5)]
+        [InlineData(typeof(LongGraphType), 1.5)]
+        [InlineData(typeof(ULongGraphType), 1.5)]
+        [InlineData(typeof(BigIntGraphType), 1.5)]
+        public void serialize_other_fail(Type graphType, object value)
+        {
+            var g = (ScalarGraphType)graphType.GetConstructor(Type.EmptyTypes).Invoke(null);
+            Should.Throw<Exception>(() => g.Serialize(value));
```

Comment:
Let's test exact exception type - `InvalidOperation` or `Format` via one more method parameter.

Reply:
No.  This is a general catch-all for some different error conditions that I can throw in a big list.  If we want individual checks, they should be in individual xxxGraphTypeTests file.  I have already updated many checks there each with different type checking capability.

Reply:
OK